core/math/math_funcs.h

```diff
@@ -178,6 +178,8 @@ class Math {
 		return abs(a - b) < CMP_EPSILON;
 	}
 
+	static _ALWAYS_INLINE_ bool is_whole_approx(real_t x) { return is_equal_approx(x, floor(x)); }
```

Comment:
As I mentioned on IRC, this should probably be using `round(x)` instead of `floor(x)`. But the question remains for the `(int)` casting in `core/variant_op.cpp`.